build/aten/src/ATen/Declarations.yaml

```diff
@@ -4156,12 +4155,33 @@
     name: window_length
     type: int64_t
   - annotation: null
-    default: '{}'
-    dynamic_type: TensorOptions
-    is_nullable: false
+    default: c10::nullopt
+    dynamic_type: ScalarType
+    is_nullable: true
     kwarg_only: true
-    name: options
-    type: const TensorOptions &
+    name: dtype
+    type: c10::optional<ScalarType>
+  - annotation: null
+    default: c10::nullopt
+    dynamic_type: Layout
```

Comment:
I never understood this, but, do you know what the difference is between `dynamic_type` and `type`?

Reply:
simple answer: dynamic_type == 'Layout', type == 'c10::optional<Layout>'

real answer: im not sure. there is separate logic for both cases. i assume in many cases we dont care what 'wraps' the object and just need to know what is it (for example, to figure if its a TensorOptions). But its a bit confusing.